druid-shell/src/platform/windows/error.rs

```diff
@@ -31,10 +40,42 @@ pub enum Error {
     NullHwnd,
 }
 
+#[allow(clippy::crosspointer_transmute)]
+fn hresult_description(hr: HRESULT) -> Option<String> {
+    unsafe {
+        let mut message_buffer: LPWSTR = std::ptr::null_mut();
+        let format_result = FormatMessageW(
+            FORMAT_MESSAGE_FROM_SYSTEM
+                | FORMAT_MESSAGE_ALLOCATE_BUFFER
+                | FORMAT_MESSAGE_IGNORE_INSERTS
+                | FORMAT_MESSAGE_MAX_WIDTH_MASK,
+            null(),
+            hr as u32,
+            0,
+            transmute(&mut message_buffer as *const LPWSTR),
```

Comment:
I think we can remove the `as *const LPWSTR` part. Otherwise we might as well just write `transmute(&message_buffer)`.

Probably best to have this instead:
```rust
transmute(&mut message_buffer)
```

The `mut` part doesn't seem required for the code to work, but it communicates the fact that the function will modify the pointer so we can keep the `mut`.

Reply:
Sorry, I was a bit slow with this comment. I wasn't waiting until you push the `DWORD` change, I swear! :grimacing:

Reply:
It seems to work without the `as *const LPWSTR`, but then clippy complains:

> warning: transmute from a reference to a pointer
  --> druid-shell\src\platform\windows\error.rs:55:13
   |
55 |             transmute(&mut message_buffer),
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `&mut message_buffer as *mut *mut u16 as *mut u16`
   |
   = note: `#[warn(clippy::useless_transmute)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#useless_transmute

Is that another false positive? I already turned off [clippy::crosspointer_transmute](https://rust-lang.github.io/rust-clippy/master/index.html#crosspointer_transmute) for this function in an earlier change.

Reply:
Clippy seems unhappy indeed. However ` as *mut LPWSTR` seems to satisfy clippy and is more to the truth, so maybe use that.

Reply:
Ah just listening to clippy apparantly this just works: `&mut message_buffer as *const LPWSTR as LPWSTR`, didn't know you could do that without transmute. Then I can remove the clippy ignore flags. Does that sound good?

Reply:
Great find! I was trying to get rid of `transmute` earlier but didn't know how. It looks like the following works as well:
```rust
&mut message_buffer as *mut LPWSTR as LPWSTR
```
Let's go with that if possible.

Reply:
Sounds good! I just pushed this change :)